### DISCUSSION TITLE
Does OCPP V2.0 central station gets from charge point MeterStop?

### DISCUSSION THREAD

**Discussion by ezolti** [ChargeTimeEU/Java-OCA-OCPP#325] [Q&A] [answered]
In OCPP V1.6 the central station gets StopTransactionRequest which has getMeterStop function.
In OCPP V2.0 there is a TransactionEventRequest with TransactionEventEnum Started , Ended , Updated.
How do I get the MeterStop?  when received transaction Ended.

@TVolden 
Can you please help or point to the person that may help?

**Comment by TVolden** [comment]
Hi @ezolti,

Maybe @robert-s-ubi can help.

**Comment by robert-s-ubi** [comment]
Hi @ezolti,

I would refer you to the OCPP 2.0.1 Specification, Part 2, Chapter 1.5. It specifies which fields in the TransactionEventRequest need to be populated for which kind of transaction event (started, updated or ended).

To answer your specific question: TransactionEventRequest#getMeterValue()

**Reply by ezolti** [reply]
Thanks, @robert-s-ubi for your answer
TransactionEventRequest.getMeterValue() returns an array of MeterValue, each one has a MeasurandEnum and Value
There is a long list of MeasurandEnum (eu.chargetime.ocpp.v201.model.types)
Which one is the one that says the same as MeterStop says in V1.6?

**Reply by ezolti** [reply]
@TVolden , @robert-s-ubi 
Please help me to figure it out

**Reply by TVolden** [reply]
Hi @ezolti,

I don't know. I would guess that this is something that is addressed in the official specification by OCA.

**Comment by robert-s-ubi** [comment, marked as answer]
Hi @ezolti,

this is really not an issue with the library. Please consult the OCPP 2.0.1 specification to understand how transaction meter values work in OCPP 2.0.1.

In short: OCPP 2.0.1 no longer has the "hardwired" MeterStop reading that OCPP 1.6 had. Instead, the meter values are fully configurable. So the ones you find in the TransactionEventRequest(ENDED) are the ones that have been configured. To get the transaction end readings, filter for the SampledValues with context = "Transaction.End".

**Reply by ezolti** [reply]
Thanks, @robert-s-ubi for your answer.
